cranelift-codegen/meta/src/cdsl/cpu_modes.rs

```diff
@@ -0,0 +1,84 @@
+use crate::cdsl::types::LaneType;
+use crate::cdsl::xform::{TransformGroup, TransformGroupIndex, TransformGroups};
+
+use std::collections::{HashMap, HashSet};
+use std::iter::FromIterator;
+
```

Comment:
While I believe I understand this code, and I understand why CpuMode makes sense at the use site of this code over in the ISA definitions, the name "CpuMode" throws me off because none of the contents seem to have to do with CpuMode (and there is no corresponding name in the python dsl). I can't think of a name for this module that is better - LegalizationScheme is the best thing I could come up with.

Reply:
Good question; it's actually something I've slightly adapted from the Python code, where there are many indirections to iterate through all the XForms (go through the TargetIsa, then each CpuMode to register all the XForms in a local TargetIsa attribute, that is then iterated on). I propose to keep it like this until encodings/binemit are ported too, to try to keep it closer to the Python code and reduce porting confusion, and then I'm definitely happy to rename this / separate it into a different data structure.